### PR TITLE
gh-95065: PoC: Argument Clinic: Support multiple '* [from ...]' deprecations

### PR DESCRIPTION
**Pull request by erlend-aasland**
This is a **rough draft**! Proof-of-concept only.

* Issue: gh-95065

### CODE REVIEW

File: Lib/test/clinic.test.c [hunk 1]
```diff
@@ -5520,6 +5520,7 @@ test_deprecate_positional_pos1_len1_optional(PyObject *module, PyObject *const *
     PyObject *a;
     PyObject *b = Py_None;
 
+    // Emit compiler warnings when we get to Python 3.14.
```
Comment (AlexWaygood) [inline Lib/test/clinic.test.c:5523]: Whatever we decide w.r.t. the larger feature, the addition of these comments to the generated code seems like a nice enhancement by itself

**Review by AlexWaygood — COMMENTED** [review]
Can we think of any current or recent situations where a stdlib function is undergoing simultaneous deprecations for  several parameters, with different removal dates for the different parameters? If this is isn't something that actually comes up that frequently, I'm inclined to say that the feature _probably_ isn't worth the extra complexity.

**Comment by erlend-aasland** [comment]
Taking sqlite3.connect as an example, I'd much rather do all deprecations in one go, rather than spreading it out over years and years with deprecation warnings.

Combining different kinds of deprecations (deprecated positional plus deprecated keyword) is another story, though, but that's OT for this PR.

**Comment by erlend-aasland** [comment]
I'm abandoning this effort. There is no reason to follow this up unless there is a pressing need for it.

**Comment by serhiy-storchaka** [comment]
I am currently working on `/ [from ...]`. It requires some rewriting to support two different deprecation terms (I will simplify the current compiler warnings for this). But after merging it, it will be easier to add support of multiple `* [from ...]` later.

In any case this feature is not necessary. We can just defer the change and turn parameters to keyword-only on the same later term. Nobody died from prolonging the deprecation period (I hope so).

**Pull request closed without merging**